src/forms-opportunities/handler.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import RUMAPIClient from '@adobe/spacecat-shared-rum-api-client';
+import { getRUMDomainkey } from '../support/utils.js';
+import { AuditBuilder } from '../common/audit-builder.js';
+import { wwwUrlResolver } from '../common/audit.js';
+
+const DAILY_THRESHOLD = 200;
+const INTERVAL = 7; // days
+const FORMS_OPPTY_QUERIES = [
+  'cwv',
+  'form-vitals',
+];
+
+export async function formsAuditRunner(auditUrl, context, site) {
+  const rumAPIClient = RUMAPIClient.createFrom(context);
+  const domainkey = await getRUMDomainkey(site.getBaseURL(), context);
+  const options = {
+    domain: auditUrl,
+    domainkey,
+    interval: INTERVAL,
+    granularity: 'hourly',
+  };
+
+  const formsAuditLinks = await rumAPIClient.queryMulti(FORMS_OPPTY_QUERIES, options);
+  const cwvMap = new Map(
+    formsAuditLinks.cwv.map((cwv) => [cwv.url, cwv]),
+  );
+
+  const auditResult = {
+    formVitals: formsAuditLinks['form-vitals'].filter((data) => {
+      // Calculate the sum of all values inside the `pageview` object
+      // eslint-disable-next-line max-len
+      const pageviewsSum = Object.values(data.pageview).reduce((sum, value) => sum + value, 0);
+      return pageviewsSum >= DAILY_THRESHOLD * INTERVAL;
+    })
+      .map((formVital) => {
+        const cwvData = cwvMap.get(formVital.url);
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const { url, pageviews, ...filteredCwvData } = cwvData || {};
+        return {
+          ...formVital,
+          cwv: filteredCwvData, // Append cwv data
```

Comment:
can we somehow avoid to persist the CWV data from RUM and reuse the data that's already persisted for sites?

Reply:
@nit23uec leaving this as an information depending on which direction you guys would like to go: you can get the latest cwv result from dynamoDB directly using `dataAccess.getLatestAuditForSite(siteId, 'cwv')' which returns the latest cwv audits result. Bare in mind, CWV audit is weekly.

Reply:
I think its better to run the query to be more accurate, if they use the `dataAccess.getLatestAuditForSite(siteId, 'cwv')`, the data could be a day old depending on when this forms audit will run relative to cwv audit.

Reply:
@05rahulsingh @vdua rather than adding the formVital and cwv data to the audit, wouldn't it make sense to find the forms that have lower engagement and lower cwv performanceScore?

Reply:
That is something which we will do it in spacecat-reporting-worker